src/components/auth/Register.jsx

```diff
@@ -1,9 +1,332 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Link } from 'react-router-dom';
+import Modal from 'react-modal';
+import { registerStudent, registerTeacher } from '../../util/DataBaseRequests';
+import FormInput from '../common/FormInput';
+import FormSubmitBtn from '../common/FormSubmitBtn';
+import FormErrorMsg from '../common/FormErrorMsg';
+
 const Register = () => {
+  const [userRole, setUserRole] = useState('student');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [dateOfBirth, setDateOfBirth] = useState('');
+  const [isAdultNameRequired, setIsAdultNameRequired] = useState(false);
+  const [adultName, setAdultName] = useState('');
+  const [adultNameError, setAdultNameError] = useState('');
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [confirmEmail, setConfirmEmail] = useState('');
+  const [confirmEmailError, setConfirmEmailError] = useState('');
+  const [password, setPassword] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const navigate = useNavigate();
+
+
+  const handleUserRegistration = async (event) => {
```

Comment:
`handleUserRegistration` is doing a lot. I suggest separating some concerns here. We have form validation and user registration.

Reply:
All form validation logic was moved to a separate function